src/main/java/com/squareup/catalog/demo/Main.java

```diff
@@ -164,13 +170,45 @@ void processArgs(String[] args) {
       return;
     }
 
-    // Configure OAuth2 access token for authorization: oauth2
-    OAuth oauth2 = (OAuth) apiClient.getAuthentication("oauth2");
-    oauth2.setAccessToken(accessToken);
+    Builder apiClientBuilder = new SquareClient.Builder();
+
+    if (environment != null && !environment.equalsIgnoreCase("sandbox")
+        && !environment.equalsIgnoreCase("production")) {
+      // was set to something that we do not support.
+      logger.error("If you choose to use the -env flag, you must either specify \"sandbox\" or \"production\"");
+      return;
+    }
+
+    // Decide on environment.
+    // If both enviroment and base-url were set, the environment choice will
+    // override.
+    Environment env;
+    if (environment == null) {
+      // if environment was not set, check if base url was provided. If so, set
+      // environment to custom.
+      // Otherwise, set environment to be sandbox by default.
+      if (customUrl != null) {
+        env = Environment.CUSTOM;
+        apiClientBuilder.customUrl(customUrl);
+      } else {
+        env = Environment.SANDBOX;
+      }
+    } else {
+      env = environment.equalsIgnoreCase("sandbox") ? Environment.SANDBOX : Environment.PRODUCTION;
+    }
+
+    // Build the client using the arguments provided
+    SquareClient apiClient = apiClientBuilder.environment(env).accessToken(accessToken).build();
+
+    CatalogApi catalogApi = apiClient.getCatalogApi();
+    LocationsApi locationsApi = apiClient.getLocationsApi();
+
+    setCurrencyAcrossApplication(locationsApi);
 
-    CatalogApi catalogApi = new CatalogApi(apiClient);
-    LocationsApi locationsApi = new LocationsApi(apiClient);
     executeExample(command, cleanup, catalogApi, locationsApi);
+
+    // Necessary in order for the program not to hang (kill backround threads).
+    System.exit(0);
```

Comment:
since we have this, why is it still necessary to use ctrl+c to kill the program?